app/Http/Controllers/Assets/AssetCheckinController.php

```diff
@@ -116,6 +116,12 @@ public function store(AssetCheckinRequest $request, $assetId = null, $backto = n
             $seat->update(['assigned_to' => null]);
         });
 
+        $settings = \App\Models\Setting::getSettings();
+
+        if($settings->require_checkinout_notes=="1" && (is_null($request->note))) {
+            return redirect()->to("hardware/$assetId/checkin")->with('error', trans('admin/hardware/message.update.no_note'));
+        }
```

Comment:
Let's move this to the form request for this method, `AssetCheckinRequest`. It'll get added to `$rules`.

Since there are no rules there now the method would look something like

```
rules = [];

if $setting->require_checkinout_notes
	rules['notes'] = 'required|string|maybe-something-else';

return rules;
```

Reply:
would we also want to do this on the `AssetCheckoutRequest`?

Reply:
Yup